### PR TITLE
Turn on enableSuspense in fbsource and test-renderer.fb

### PR DESCRIPTION
**Pull request by cyan33**
To move forward with the suspense code splitting in fbsource, we need to turn on the `enableSuspense` feature flag, also that in test-renderer in fbsource.

I've checked the bundle in the `build` folder. Now it's on in:

- `ReactFabric-dev.js`
- `ReactFabric-prod.js`
- `ReactNativeRenderer-dev.js`
- `ReactNativeRenderer-prod.js`

But there is no corresponding bundle for test-renderer, like we have `ReactTestRenderer-dev.js` in `build/facebook-www`. I'm wondering if I did it wrong. Can someone help double check on this?

### CODE REVIEW

**Comment by react-linter** [comment]
<details>
  <summary>Details of bundled changes.</summary>

  <p>Comparing: e79366d5497e669af462b705303bee7edef02585...1b04bd8fc2c14a9e99b74be50985a4601ae40261</p>


  
## react-native-renderer
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
ReactNativeRenderer-dev.js | -0.0% | -0.0% | 478.31 KB | 478.29 KB | 105.71 KB | 105.71 KB | RN_FB_DEV
**ReactNativeRenderer-prod.js** | **-0.3%** | **-0.2%** | **212.26 KB** | **211.66 KB** | **36.92 KB** | **36.84 KB** | **RN_FB_PROD**
ReactFabric-dev.js | -0.0% | 0.0% | 468.53 KB | 468.53 KB | 103.29 KB | 103.29 KB | RN_FB_DEV
**ReactFabric-prod.js** | **:small_red_triangle:+4.6%** | **:small_red_triangle:+4.2%** | **193.03 KB** | **201.83 KB** | **33.76 KB** | **35.18 KB** | **RN_FB_PROD**
ReactNativeRenderer-profiling.js | -0.3% | -0.2% | 215.35 KB | 214.74 KB | 37.61 KB | 37.54 KB | RN_FB_PROFILING
ReactFabric-profiling.js | +4.5% | +4.1% | 195.8 KB | 204.59 KB | 34.36 KB | 35.76 KB | RN_FB_PROFILING

  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by gaearon** [comment]
Can we keep it enabled in all FB builds of test renderer, and not just for fbsource? I'm not a fan of making test renderer builds different for fbsource and www. Too many configurations.

**Comment by bvaughn** [comment]
I like Dan's suggestion.

In case it's unclear, we'd need to add a new feature flag file to do this (e.g. `ReactFeatureFlags.test-renderer.native-fb.js`) and add that entry to the `forks` file for `entry === 'react-test-renderer'` and `bundleType === RN_FB_*`

**Comment by bvaughn** [comment]
Sorry, my initial comment was crazy and confusing, so I deleted it. 😁

> But there is no corresponding bundle for test-renderer, like we have `ReactTestRenderer-dev.js` in `build/facebook-www`. I'm wondering if I did it wrong. Can someone help double check on this?

In fbsource, we install `react-test-renderer` directly from NPM. If you look inside of `RKJSModules/Libraries/Packages/react-test-renderer/ReactTestRenderer` you'll see a forwarding file that just points to the NPM package.

As far as imports go, fbsource has a mix of direct `react-test-renderer` imports and `ReactTestRenderer` forwarded imports. (The direct ones are mostly in open source Relay code.)

Not sure what the best option is here. Need to think about it a bit more.

**Comment by bvaughn** [comment]
> Also, in fbsource, if we're using test-renderer from npm, does that mean we need to turn that on for open source?

Currently, yes. We would have to turn the flag on for open source _and do an actual open source release_ in order to use it. (We could get away with doing an alpha release perhaps.)

I don't think we want to turn the feature on for any renderers in open source until we're ready to turn it on for _all_ renderers though.

File: scripts/rollup/forks.js [hunk 1]
```diff
@@ -75,10 +75,13 @@ const forks = Object.freeze({
         return 'shared/forks/ReactFeatureFlags.persistent.js';
       case 'react-test-renderer':
         switch (bundleType) {
+          case RN_FB_DEV:
+          case RN_FB_PROD:
+          case RN_FB_PROFILING:
           case FB_WWW_DEV:
           case FB_WWW_PROD:
           case FB_WWW_PROFILING:
-            return 'shared/forks/ReactFeatureFlags.test-renderer.www.js';
```
Comment (bvaughn) [inline scripts/rollup/forks.js]: This rename might not be a great idea. Our current naming convention, although confusing, kind fo seems to be that "fb" stands for Facebook's "fbsource" and "www" stands for www.

We _could_ do a cleanup pass where we made them more cohesive (i.e. either "facebook", "facebook-www", or "facebook-fbsource") but unless we do a larger cleanup like that, I think the www feature flag file makes better sense with its original name.

**Comment by cyan33** [comment]
Got it. So in my understanding if we want to turn on the feature flag of test-renderer in fbsource, we should do either of these two options:

1. Turn that feature flag on in open source, and do an actual release, which is not recommended.
2. Use our own fork of react-test-renderer rather than from npm in fbsource.

Let's just turn that on for only React in fb-source. Maybe I'll add a follow up task to add test case.

**Comment by bvaughn** [comment]
> Let's just turn that on for React in fb-source first. Maybe I'll add a follow up task to add some test cases.

Won't this prevent us from running Relay tests for the new suspense query renderer? (Relay is developed in fbsource first and synced to www, I believe.)

**Comment by cyan33** [comment]
The SuspenseQueryRenderer is currently in `shared/relay/www` as a forked module to enable quick iteration. So I think it shouldn't matter in this case.

**Comment by bvaughn** [comment]
Ah, so the suspense one only lives in www for the time being? Cool.

**Comment by bvaughn** [comment]
I'm still not sure about this PR though.

It would mean that Facebook native apps can use suspense in React Native / React Fabric– but any tests that happen to include components with suspense will fail. I don't think we can safely land that.

**Comment by cyan33** [comment]
That feature flag should be turned on internally in fbsource. Thanks for all the help @bvaughn @gaearon . Closed this PR.

**Pull request closed without merging**